clrtrust

```diff
@@ -84,6 +84,143 @@ find_all_certs() {
+                opt_verbose=1
+                shift
+                ;;
+            *)
+                print_check_help
+                return $EINVAL
+                ;;
+        esac
+    done
+
+    if [ ! -z "${VERBOSE}" ]; then
+        opt_verbose=1
+    fi
+
+    if [ -e "${CLR_LOCAL_TRUST_SRC}" ]; then
+        if [ ! -d "${CLR_LOCAL_TRUST_SRC}" ]; then
+            1>&2 echo "${CLR_LOCAL_TRUST_SRC} must be a directory."
+            test ! -z $opt_verbose && 1>&2 cat <<EOF
+    ${CLR_LOCAL_TRUST_SRC} is used by ${BASENAME} to store local (user-added) trust and
+    distrust data.
+EOF
+            ret=$EBADST
+        fi
+        if [ -e "${CLR_LOCAL_TRUST_SRC}/distrusted" -a ! -d "${CLR_LOCAL_TRUST_SRC}/distrusted" ]; then
+            1>&2 echo "$CLR_LOCAL_TRUST_SRC/distrusted must be a directory."
+            test ! -z $opt_verbose && 1>&2 cat <<EOF
+    ${CLR_LOCAL_TRUST_SRC}/trusted is a directory used by ${BASENAME} to store
+    trusted root CA certificates.
+EOF
+            ret=$EBADST
+        fi
+        if [ -e "${CLR_LOCAL_TRUST_SRC}/trusted" -a ! -d "${CLR_LOCAL_TRUST_SRC}/trusted" ]; then
+            1>&2 echo "$CLR_LOCAL_TRUST_SRC/trusted must be a directory."
+            test ! -z $opt_verbose && 1>&2 cat <<EOF
+    ${CLR_LOCAL_TRUST_SRC}/distrusted is a directory used by ${BASENAME} to store
+    distrusted root CA certificates.
+EOF
+            ret=$EBADST
+        fi
+    fi # it's OK if CLR_LOCAL_TRUST_SRC does not exist
+
+    if [ -e "${CLR_CLEAR_TRUST_SRC}" ]; then
+        for dir in "${CLR_CLEAR_TRUST_SRC}" "${CLR_CLEAR_TRUST_SRC}/trusted"; do
+            if [ ! -d "${dir}" ]; then
+                1>&2 echo "${dir} must be a directory."
+                test ! -z $opt_verbose && 1>&2 cat <<EOF
+    ${dir} is a directory which contains trust and distrust data provided by
+    Clear Linux. It is installed initially and updated by swupd. Remove ${dir}
+    and run 'swupd verify --fix' to reinstall.
+EOF
+                ret=$EBADST
+            fi
+        done
+    else # it's not OK if CLR_CLEAR_TRUST_SRC does not exist
+        1>&2 echo "${CLR_CLEAR_TRUST_SRC} does not exist."
+        test ! -z $opt_verbose && 1>&2 cat <<EOF
+    ${CLR_CLEAR_TRUST_SRC} is a directory which contains trust and distrust data
+    provided by Clear Linux. It is installed initially and updated by swupd. Run
+    'swupd verify --fix' to reinstall.
+EOF
+        ret=$EBADST
+    fi
+
+    dir=$(dirname ${CLR_TRUST_STORE})
+    if [ ! -e "$dir" ]; then
+        1>&2 echo "${dir} does not exit."
+    fi
+
+    if [ "x${CLR_TRUST_STORE}" = "x${CLR_TRUST_STORE_DFLT}" ]; then
+        usr="root"
+    else
+        usr=${USER}
+    fi
+
+    stat=($(stat -L -c "0%a %G %U" "$dir"))
+    ownusr=${stat[2]}
+    owngrp=${stat[1]}
+    perm=${stat[0]}
+
+    if [ -z "$perm" -o -z "$ownusr" -o -z "$owngrp" -o -z "$usr" ]; then
+        return $EERR
+    fi
+
+    if [ "x$ownusr" = "x$usr" ] && (( $perm & 0200 )); then
```

Comment:
This seems a rather verbose way to do a ```test -w "$dir"```.

Reply:
Well, this entire `stat` exercise is because I need to check if it's writeable for `root` when writing to the default location or for the current user if the location is overriden (by the environment var). So simple checks will not suffice.